cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -184,4 +220,135 @@ public Integer apply(DatasetContext<Table> ctx) throws Exception {
   public interface DeleteCallback {
     public void handle(Location location);
   }
+
+  /**
+   * Upgrades the log meta table
+   * Note: Currently this supports upgrade from 2.6 to 2.8 and does upgrade for namepspaces
+   *
+   * @throws Exception
+   */
+  public void upgrade(final DatasetFramework dsFramework) throws Exception {
```

Comment:
upgrade method shouldn't have to take a dataset framework. Lets pass it and a Cconf in the constructor instead of a LogSaverTableUtil.